packages/main/src/Menu.js

```diff
@@ -0,0 +1,490 @@
+import {
+	getI18nBundle,
+} from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+import ResponsivePopover from "./ResponsivePopover.js";
+import Button from "./Button.js";
+import List from "./List.js";
+import StandardListItem from "./StandardListItem.js";
+import Icon from "./Icon.js";
+
+import staticAreaMenuTemplate from "./generated/templates/MenuTemplate.lit.js";
+
+import {
+	MENU_BACK_BUTTON_ARIA_LABEL,
+	MENU_CLOSE_BUTTON_ARIA_LABEL,
+} from "./generated/i18n/i18n-defaults.js";
+
+// Styles
+import staticAreaMenuCss from "./generated/themes/Menu.css.js";
+
+/**
+ * @public
+ */
+const metadata = {
+	tag: "ui5-menu",
+	properties: /** @lends sap.ui.webcomponents.main.Menu.prototype */ {
+		/**
+		 * Defines the title of the menu (displayed on mobile).
+		 * @type {string}
+		 * @defaultvalue ""
+		 * @public
+		 */
+		 title: {
+			type: String,
+		},
+
+		/**
+		 * Defines if the menu is sub-menu (not first-level).
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @private
+		 */
+		_isSubMenu: {
+			type: Boolean,
+			noAttribute: true,
+		},
+
+		/**
+		 * Stores parent menu item (if there is such).
+		 * @type {object}
+		 * @private
+		 */
+		 _parentMenuItem: {
+			type: Object,
+		},
+
+		/**
+		 * Stores menu item that have sub-menu opened.
+		 * @type {object}
+		 * @private
+		 */
+		 _openedSubMenuItem: {
+			type: Object,
+		},
+
+		/**
+		 * Stores id of a list item that opened sub-menu.
+		 * @type {string}
+		 * @private
+		 */
+		 _subMenuOpenerId: {
+			type: String,
+		},
+
+		/**
+		 * Defines the currently available menu items.
+		 * (in case of non-phone devices these are the items of the menu,
+		 * but for phone devices the items of the currently opened sub-menu
+		 * will be populated here)
+		 * @type {array}
+		 * @private
+		 */
+		 _currentItems: {
+			type: Object,
+			multiple: true,
+		},
+
+		/**
+		 * Stores a list of parent menu items for each sub-menu (on phone).
+		 * @type {array}
+		 * @private
+		 */
+		 _parentItemsStack: {
+			type: Object,
+			multiple: true,
+		},
+
+		/**
+		 * Stores a responsive popover when it is opened.
+		 * @type {object}
+		 * @private
+		 */
+		_popover: {
+			type: Object,
+		},
+	},
+	managedSlots: true,
+	slots: /** @lends sap.ui.webcomponents.main.Menu.prototype */ {
+		/**
+		 * Defines the items of this component.
+		 *
+		 * @type {sap.ui.webcomponents.main.MenuItem[]}
+		 * @slot items
+		 * @public
+		 */
+		 "default": {
+			propertyName: "items",
+			type: HTMLElement,
+			invalidateOnChildChange: true,
+		},
+	},
+	events: /** @lends sap.ui.webcomponents.main.Menu.prototype */ {
+		/**
+		 * Fired when an item is being selected.
+		 *
+		 * @event sap.ui.webcomponents.main.Menu#item-selected
+		 * @param {object} item The currently selected menu item.
+		 * @param {string} text The text of the currently selected menu item.
+		 * @public
+		 */
+		 "item-selected": {
```

Comment:
Is it more like "clicked" than "selected", the menu item does not have selection state

Reply:
This is somehow correct, and not... Yes, there is no selected state during the interaction with the menu, but at last if someone clicks on an item, this item is "selected" and the menu is closed... anyway, changed to 'item-clicked'...